pkg/scheduler/factory_test.go

```diff
@@ -67,208 +64,299 @@ func TestCreate(t *testing.T) {
-		"predicates" : [
-			{"name" : "TestZoneAffinity", "argument" : {"serviceAffinity" : {"labels" : ["zone"]}}},
-			{"name" : "TestRequireZone", "argument" : {"labelsPresence" : {"labels" : ["zone"], "presence" : true}}},
-			{"name" : "PodFitsResources"},
-			{"name" : "PodFitsHostPorts"}
-		],
-		"priorities" : [
-			{"name" : "RackSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "rack"}}},
-			{"name" : "NodeAffinityPriority", "weight" : 2},
-			{"name" : "ImageLocalityPriority", "weight" : 1},
-			{"name" : "InterPodAffinityPriority", "weight" : 1}
-		],
-		"hardPodAffinitySymmetricWeight" : 10
-	}`)
-	if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), configData, &policy); err != nil {
-		t.Fatalf("Invalid configuration: %v", err)
-	}
-	if _, err := factory.createFromConfig(policy); err != nil {
-		t.Fatal(err)
-	}
-	// TODO(#87703): Verify that the entire pluginConfig is correct.
-	foundAffinityCfg := false
-	for _, cfg := range factory.profiles[0].PluginConfig {
-		if cfg.Name == interpodaffinity.Name {
-			foundAffinityCfg = true
-			wantArgs := &schedulerapi.InterPodAffinityArgs{HardPodAffinityWeight: 10}
-
-			if diff := cmp.Diff(wantArgs, cfg.Args); diff != "" {
-				t.Errorf("wrong InterPodAffinity args (-want, +got): %s", diff)
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			recorderFactory := profile.NewRecorderFactory(events.NewBroadcaster(&events.EventSinkImpl{Interface: client.EventsV1()}))
+
+			_, err := New(
+				client,
+				informerFactory,
+				recorderFactory,
+				make(chan struct{}),
+				WithAlgorithmSource(createAlgorithmSourceFromPolicy(tc.configData, client)),
+				WithBuildFrameworkCapturer(func(p schedulerapi.KubeSchedulerProfile) {
+					if p.SchedulerName != v1.DefaultSchedulerName {
+						t.Errorf("unexpected scheduler name (want %q, got %q)", v1.DefaultSchedulerName, p.SchedulerName)
+					}
+
+					if diff := cmp.Diff(tc.wantPluginConfig, p.PluginConfig); diff != "" {
+						t.Errorf("unexpected plugins config diff (-want, +got): %s", diff)
+					}
+
+					if diff := cmp.Diff(tc.wantPlugins, p.Plugins); diff != "" {
+						t.Errorf("unexpected plugins diff (-want, +got): %s", diff)
+					}
+				}),
+			)
+
+			if err != nil {
+				t.Fatalf("Error constructing: %v", err)
 			}
-		}
-	}
-	if !foundAffinityCfg {
-		t.Errorf("args for InterPodAffinity were not found")
-	}
-}
-
-func TestCreateFromEmptyConfig(t *testing.T) {
-	var configData []byte
-	var policy schedulerapi.Policy
-
-	client := fake.NewSimpleClientset()
-	stopCh := make(chan struct{})
-	defer close(stopCh)
-	factory := newConfigFactory(client, stopCh)
-
-	configData = []byte(`{}`)
-	if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), configData, &policy); err != nil {
-		t.Errorf("Invalid configuration: %v", err)
-	}
-
-	if _, err := factory.createFromConfig(policy); err != nil {
-		t.Fatal(err)
-	}
-	prof := factory.profiles[0]
-	wantConfig := []schedulerapi.PluginConfig{}
```

Comment:
Removing this test because the lack of policy content is already being tested in the first matrix test:

```
"policy with unspecified predicates or priorities uses default"
```